apps/studio/data/custom-domains/check-cname-mutation.ts

```diff
@@ -22,19 +23,24 @@ export type CheckCNAMERecordResponse = {
 
 // [Joshen] Should tally with https://github.com/supabase/cli/blob/63790a1bd43bee06f82c4f510e709925526a4daa/internal/utils/api.go#L98
 export async function checkCNAMERecord({ domain }: CheckCNAMERecordVariables) {
-  const res = await fetchHandler(`https://one.one.one.one/dns-query?name=${domain}&type=CNAME`, {
-    method: 'GET',
-    headers: { accept: 'application/dns-json' },
-  })
-  const verification = (await res.json()) as CheckCNAMERecordResponse
+  try {
+    const headers = await constructHeaders({ 'Content-Type': 'application/json' })
+    const res: CheckCNAMERecordResponse = await fetchHandler(`${BASE_PATH}/api/check-cname`, {
+      headers,
+      method: 'POST',
```

Comment:
nit: since the underlying endpoint is just a `GET` request, maybe it's worth making this a `GET` and sending the domain as a query param?